server/db/models/MovieQueues.js

```diff
@@ -27,6 +27,65 @@ var schema = new mongoose.Schema({
 })
 
 
+var shiftQueue = function(user) {
+	for(var i = 0; i < user.queue.length; i++) {
+		if(user.queue[i].status === 'pending') {
+			user.queue[i].status = 'active';
+			user.queue[i].priority--;
+			break;
+		}
+	}
+	return user;
+}
+
+var checkStatus = function(user) {
+	var count = 0;
+	user.queue.forEach(function(item) {
+		if(item.status === 'active') {
+			count++;
+		}
+	})
+	return count;
+}
+
+schema.methods.addToQueue = function(movieId) {
+	var user = this;
```

Comment:
I think there's maybe something like `this.constructor` that can get you the "parent" model of a ref